src/app/charts/chart.component.ts

```diff
@@ -115,18 +116,15 @@ export class ChartComponent implements OnChanges, OnDestroy, AfterViewInit {
         this.cancelDataRequest();
     }
 
-    updateChart(extraParams: any) {
+    updateChart(extraParams: IndicatorParams) {
         this.cancelDataRequest();
         this.chartData = [];
         this.rawChartData = [];
 
```

Comment:
I'm not understanding why the typecheck doesn't complain when ThresholdIndicatorParams are sent through

Reply:
Because `interface ThresholdIndicatorParams` extends `interface `IndicatorParams`

Reply:
Ah interesting. Extending works the opposite direction I was expecting, then. Cool.